mewarpx/scripts_ancillary/tinyprofiler_plotting_example.py

```diff
@@ -0,0 +1,270 @@
+'''
+This script shows 3 examples of plotting the json data generated:
+- Plotting a stacked bar graph of the max/avg percents of functions for multiple runs.
+- Plotting the profiling data of a union of the top n expensive functions for multiple runs.
+- Plotting the profiling data of a specified collection of functions for multiple runs.
+
+As well as converting all the .out files in a directory to JSON files.
+'''
+
+import hatchet as ht
+import json
+import numpy as np
+import os
+import math
+from pathlib import Path
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.pyplot import cm
+
+from mewarpx.utils_store import profileparser
+
+
+def get_parameter_name(exclusive, average):
+    parameter = ""
+    if exclusive:
+        parameter += "excl_"
+    else:
+        parameter += "incl_"
+
+    if average:
+        parameter += "avg"
+    else:
+        parameter += "max"
+
+    return parameter
+
+def get_description(exclusive, average):
+    description = ""
+    if exclusive:
+        description += "exclusive "
+    else:
+        description += "inclusive "
+
+    if average:
+        description += "average"
+    else:
+        description += "max"
+
+    return description
+
+def plot_max_percents(exclusive):
+    dataframes = []
+    parameter = "excl_max_percent" if exclusive else "incl_max_percent"
+    description = "Exclusive" if exclusive else "Inclusive"
+    for name in Path("warpx_outputs/").rglob("*.json"):
+        print("plotting:", name)
+
+        f = open(name)
+        data = json.load(f)
```

Comment:
```suggestion
        with open(name) as f:
            data = json.load(f)
```
This with block auto-closes f when it goes out of scope; much better than accidentally forgetting to close it!